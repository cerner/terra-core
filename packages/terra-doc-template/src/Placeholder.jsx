import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames/bind';
import styles from './Placeholder.module.scss';

const cx = classNames.bind(styles);

const propTypes = {
  /**
   * The themed variant of the text and border. Example, use the light variant when against a dark background.
   */
  variant: PropTypes.oneOf(['light', 'dark']),
  /**
   * The placeholder text to be displayed.
   */
  title: PropTypes.string,
};

const defaultProps = {
  variant: 'dark',
  title: '',
};

const Placeholder = ({
  variant,
  title,
  ...customProps
}) => {
  const placeholderClassNames = cx([
    'placeholder',
    customProps.className,
  ]);

  const innerClassNames = cx([
    'inner',
    `is-${variant}`,
  ]);

  return (
    <div {...customProps} className={placeholderClassNames}>
      <div className={innerClassNames}>
        <h2>{title}</h2>
      </div>
    </div>
  );
};

Placeholder.propTypes = propTypes;
Placeholder.defaultProps = defaultProps;

export default Placeholder;
