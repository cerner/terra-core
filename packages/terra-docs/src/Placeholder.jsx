import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';
import classNamesBind from 'classnames/bind';
import ThemeContext from 'terra-theme-context';
import styles from './Placeholder.module.scss';

const cx = classNamesBind.bind(styles);

const propTypes = {
  /**
   * The placeholder text to be displayed.
   */
  title: PropTypes.string,
};

const defaultProps = {
  title: '',
};

const Placeholder = ({
  title,
  ...customProps
}) => {
  const theme = React.useContext(ThemeContext);
  const placeholderClassNames = classNames(
    cx([
      'placeholder',
      theme.className,
    ]),
    customProps.className,
  );

  const innerClassNames = cx([
    'inner',
  ]);

  return (
    <div {...customProps} className={placeholderClassNames}>
      <div className={innerClassNames}>
        <p>{title}</p>
      </div>
    </div>
  );
};

Placeholder.propTypes = propTypes;
Placeholder.defaultProps = defaultProps;

export default Placeholder;
