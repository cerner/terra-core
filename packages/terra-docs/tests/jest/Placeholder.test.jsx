import React from 'react';
import Placeholder from '../../src/Placeholder';

describe('Placeholder', () => {
  // Snapshot Tests
  it('should render a default component', () => {
    const wrapper = shallow(<Placeholder />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should show with a title', () => {
    const wrapper = shallow(<Placeholder title="test" />);
    expect(wrapper).toMatchSnapshot();
  });
});
