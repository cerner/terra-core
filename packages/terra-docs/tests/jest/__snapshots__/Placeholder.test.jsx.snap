// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Placeholder should render a default component 1`] = `
<div
  className="placeholder"
>
  <div
    className="inner"
  >
    <p />
  </div>
</div>
`;

exports[`Placeholder should show with a title 1`] = `
<div
  className="placeholder"
>
  <div
    className="inner"
  >
    <p>
      test
    </p>
  </div>
</div>
`;
